src/main/java/com/direwolf20/buildinggadgets/common/config/Config.java

```diff
@@ -0,0 +1,176 @@
+package com.direwolf20.buildinggadgets.common.config;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.config.Config.*;
+
+import javax.annotation.Nonnull;
+
+//hardcode this name, so that we don't use a different config file than before we used Annotations
+@net.minecraftforge.common.config.Config(modid = BuildingGadgets.MODID, name = "BuildingGadgets", category =  Config.CATEGORY_ROOT)
+public class Config {
+    @Nonnull
+    private static String getName(Block block) {
+        ResourceLocation name = block.getRegistryName();
+        if (name == null)
+            throw new IllegalArgumentException("A registry name for the following block could not be found: " + block);
+
+        return name.toString();
+    }
+    @Ignore
+    static final String CATEGORY_ROOT = "general";
+    @Ignore
+    private static final String LANG_KEY_ROOT = BuildingGadgets.MODID +".config."+CATEGORY_ROOT;
+    @Ignore
+    private static final String LANG_KEY_BLACKLIST = LANG_KEY_ROOT+".subCategoryBlacklist";
+    @Ignore
+    private static final String LANG_KEY_GADGETS = LANG_KEY_ROOT+".subCategoryGadgets";
+    @Ignore
+    private static final String LANG_KEY_GADGET_BUILDING = LANG_KEY_GADGETS+".gadgetBuilding";
+    @Ignore
+    private static final String LANG_KEY_GADGET_EXCHANGER = LANG_KEY_GADGETS+".gadgetExchanger";
+    @Ignore
+    private static final String LANG_KEY_GADGET_DESTRUCTION = LANG_KEY_GADGETS+".gadgetDestruction";
+    @Ignore
+    private static final String LANG_KEY_GADGET_COPY_PASTE = LANG_KEY_GADGETS+".gadgetCopyPaste";
+
+    @RangeDouble(min=1,max=48)
+    @Name("Trace Distance")
+    @Comment("Defines how far away you can build")
+    @LangKey(LANG_KEY_ROOT+".rayTraceRange")
+    public static double rayTraceRange = 32;
+
+    @Comment("Set to true for Forge Energy Support, set to False for vanilla Item Damage")
+    @LangKey(LANG_KEY_ROOT+".poweredByFE")
+    public static boolean poweredByFE = true;
+
```

Comment:
Ok, good idea...

Reply:
Though I'll loose what's left of compat with old config Files then...?

Reply:
You can do a check for the version of a config file or the lack there of and then parse the old config and update it to the new one, remove the parsing in future releases.

Reply:
There isn't much compat left anyway. If that much is changing, might as well just change everything at once. It can simply be a hard reset, where the old config files are considered obsolete.